packages/api/lambdas/message-consumer.js

```diff
@@ -12,25 +13,42 @@ const { queueMessageForRule } = require('../lib/rulesHelpers');
  * `getKinesisRules` scans and returns DynamoDB rules table for enabled,
  * 'kinesis'-type rules associated with the * collection declared in the event
  *
- * @param {Object} collection - lambda event
+ * @param {Object} queryParams - lambda event
  * @returns {Array} List of zero or more rules found from table scan
  */
-async function getKinesisRules(collection) {
+async function getKinesisRules(queryParams) {
+  const names = {
+    '#st': 'state',
+    '#rl': 'rule',
+    '#tp': 'type'
+  };
+  let filter = '#st = :enabledState AND #rl.#tp = :ruleType';
+  const values = {
+    ':enabledState': 'ENABLED',
+    ':ruleType': 'kinesis'
+  };
+  if (queryParams.collectionName) {
+    values[':collectionName'] = queryParams.collectionName;
+    names['#col'] = 'collection';
+    names['#nm'] = 'name';
+    filter += ' AND #col.#nm = :collectionName';
+  }
+  if (queryParams.collectionVersion) {
```

Comment:
It's just using whatever is in the notification. I tried to engineer it to support the CNM use case, which should have both, as well as a more general sns/kinesis use case, where `collection` is an object.

However, in the case that either does not exist, I would still like to filter as much as possible rather than bomb out or schedule for all rules as errors and logging from this function are hard to surface to operators.